packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -797,6 +785,83 @@ class RenderUnconstrainedBox extends RenderAligningShiftedBox with DebugOverflow
   }
 }
 
+/// Renders a box, imposing no constraints on its child, allowing the child to
+/// render at its "natural" size.
+///
+/// This allows a child to render at the size it would render if it were alone
+/// on an infinite canvas with no constraints. This box will then attempt to
+/// adopt the same size, within the limits of its own constraints. If it ends
+/// up with a different size, it will align the child based on [alignment].
+/// If the box cannot expand enough to accommodate the entire child, the
+/// child will be clipped.
+///
+/// In debug mode, if the child overflows the box, a warning will be printed on
+/// the console, and black and yellow striped areas will appear where the
+/// overflow occurs.
+///
+/// See also:
+///
+///  * [RenderConstrainedBox], which renders a box which imposes constraints
+///    on its child.
+///  * [RenderConstrainedOverflowBox], which renders a box that imposes different
+///    constraints on its child than it gets from its parent, possibly allowing
+///    the child to overflow the parent.
+///  * [RenderSizedOverflowBox], a render object that is a specific size but
+///    passes its original constraints through to its child, which it allows to
+///    overflow.
+class RenderUnconstrainedBox extends RenderConstraintsTransformBox {
```

Comment:
I wanted something like this when I was working on InteractiveViewer, and this is how I did it:

https://github.com/flutter/flutter/blob/c30fdfbe82d0d291c00ef199a35657990bba395c/packages/flutter/lib/src/widgets/interactive_viewer.dart#L1073-L1082

If I remember correctly it was actually really hard to do for sizes bigger than the screen.  Using a ConstrainedBox with infinite constraints didn't work, strangely.